app/services/archive_uploader.rb

```diff
@@ -0,0 +1,69 @@
+class ArchiveUploader
+  # see: https://docs.ovh.com/fr/storage/pcs/capabilities-and-limitations/#max_file_size-5368709122-5gb
+  # officialy it's 5Gb. but let's avoid to reach the exact spot of the limit
+  MAX_FILE_SIZE_FOR_BACKEND_BEFORE_CHUNKING = ENV.fetch('ACTIVE_STORAGE_FILE_SIZE_THRESHOLD_BEFORE_CUSTOM_UPLOAD') { 4.gigabytes }
+
+  def upload
+    return upload_with_active_storage if active_storage_service_local?
+
+    case File.size(filepath)
+    when 0..MAX_FILE_SIZE_FOR_BACKEND_BEFORE_CHUNKING then
+      upload_with_active_storage
+    else
+      upload_with_custom_wrapper
+    end
+  end
+
+  def active_storage_service_local?
+    Rails.application.config.active_storage.service == :local
+  end
+
+  private
+
+  attr_reader :procedure, :archive, :filepath
+
+  def upload_with_active_storage
+    archive.file.attach(
+      io: File.open(filepath),
+      filename: object_filename,
+      metadata: { virus_scan_result: ActiveStorage::VirusScanner::SAFE }
+    )
+  end
+
+  def upload_with_custom_wrapper
+    blob = ActiveStorage::Blob.create_before_direct_upload!(
+        key: SecureRandom.uuid,
```

Comment:
Openstack objectstorage permet d'organiser les fichiers par pseudo répertoire, du coup et pour faciliter les opérations est ce que tu pourrais stocker les archives sous un répertoire dédiées ? 

par exemple :

```suggestion
        key: "archives/#{SecureRandom.uuid},
```

Reply:
merci pour la suggestion ; j'ai aussi appliqué cette logique du coté des upload via active_storage pour rester homogène. J'imagine que ce pseudo repertoire pourrait nous etre utile (genre on rate le fait de netoyer les fichiers... pratique de cleaner ts les objets plus vieux que 24h)